gapminder/data.py

```diff
@@ -27,6 +29,35 @@ def load_data():
     return data
 
 
+# Caching needs hashable stuff.
+def make_stuff_hashable(func):
+    class HDict(dict):
+        def __hash__(self):
+            # DFs are not hashable. Use Keys.
+            return hash(frozenset(self.keys()))
+
+    class HList(list):
+        def __hash__(self):
+            return hash(" ".join([str(x) for x in self]))
+
+    def _make_hashable(stuff):
+        if isinstance(stuff, dict):
+            return HDict(stuff)
+        elif isinstance(stuff, list):
+            return HList(stuff)
+        return stuff
+
+    @functools.wraps(func)
+    def wrapped(*args, **kwargs):
+        args = tuple([_make_hashable(arg) for arg in args])
+        kwargs = {k: _make_hashable(v) for k, v in kwargs.items()}
+        return func(*args, **kwargs)
+
+    return wrapped
+
+
+@make_stuff_hashable
+@functools.lru_cache(maxsize=32)
```

Comment:
why not using the regular flask - cache?
```
from flask_caching import Cache


app = dash.Dash(__name__)
cache = Cache(app.server, config={
    'CACHE_TYPE': 'filesystem',
    'CACHE_DIR': 'cache'
})
```
It's nice component which can be extended from in memory, filesystem to redis...